### PR TITLE
Implement tridiag solver for 1D

### PR DESCRIPTION
**Pull request by dpgrote**
For electrostatic simulations, this replaces the Poisson solver with a tridiagonal matrix solver when running in 1D. The solvers in AMREX do not support 1D (even though they do compile), so with this PR, the electrostatic model can now be used in 1D (without requiring a Python level solver).

Todo list
- [x] Support Dirichlet boundary conditions
- [x] Support Neumann boundary conditions
- [x] Support periodic boundary conditions
- [x] Add CI test
- [x] Check operation on GPU

### CODE REVIEW

**Comment by PhilMiller** [comment]
All looks good to me.

Out of curiosity, why not put this in AMReX?

File: Source/FieldSolver/ElectrostaticSolver.cpp [hunk 1]
```diff
@@ -242,10 +242,24 @@ WarpX::AddSpaceChargeFieldLabFrame ()
     setPhiBC(phi_fp);
 
     // Compute the potential phi, by solving the Poisson equation
-    if ( IsPythonCallBackInstalled("poissonsolver") ) ExecutePythonCallback("poissonsolver");
-    else computePhi( rho_fp, phi_fp, beta, self_fields_required_precision,
-                     self_fields_absolute_tolerance, self_fields_max_iters,
-                     self_fields_verbosity );
+    if (IsPythonCallBackInstalled("poissonsolver")) {
+
+        // Use the Python level solver (user specificed)
```
Comment (roelof-groenewald) [inline Source/FieldSolver/ElectrostaticSolver.cpp]:
```suggestion
        // Use the Python level solver (user specified)
```

**Review by roelof-groenewald — APPROVED** [review]
Looks good to me too. There is a comment in the `PICMI_inputs_1d.py` example file just below the `PoissonSolver1D` declaration that states:
"This can be removed and the MLMG solver used once #3123 is addressed."
That comment can now be removed and the maybe that issue can be closed with this PR.

File: Source/FieldSolver/ElectrostaticSolver.cpp (hunk 1 above)
Comment (dpgrote) [reply Source/FieldSolver/ElectrostaticSolver.cpp]: Thanks!

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**